components/brave_new_tab_ui/containers/newTab/index.tsx

```diff
@@ -100,6 +106,22 @@ function GetBackgroundImageSrc (props: Props) {
   return undefined
 }
 
+function GetSponsoredRichMediaBackground(props: Props): SponsoredRichMediaBackgroundInfo | undefined {
+  const wallpaperData = props.newTabData.brandedWallpaper
+  if (!props.newTabData.showBackgroundImage
+      || !wallpaperData?.isSponsored
+      || wallpaperData?.type !== 'richMedia'
+      || !wallpaperData?.wallpaperImageUrl) {
+          return undefined
+      }
+  return {
+      url: wallpaperData.wallpaperImageUrl,
+      placementId: wallpaperData.wallpaperId,
+      creativeInstanceId: wallpaperData.creativeInstanceId,
+      targetUrl: wallpaperData.logo.destinationUrl
```

Comment:
do we validate at any point client-side these targetUrls are HTTPS?

Reply:
Fixed. See other comment.